app/presenters/summary/sections/client_details.rb

```diff
@@ -30,16 +30,20 @@ def answers
         ]
 
         unless post_submission_evidence?
-          answers.push(Components::FreeTextAnswer.new(
-                         :nino, applicant.nino,
-                         change_path: edit_steps_nino_path(subject: 'client'),
-                         show: true
-                       ))
-
-          answers.push(Components::FreeTextAnswer.new(
-                         :arc, applicant.arc,
-                         change_path: edit_steps_nino_path(subject: 'client'),
-                       ))
+          if applicant.has_nino == YesNoAnswer::YES.to_s
+            answers.push(Components::FreeTextAnswer.new(
+                           :nino, applicant.nino,
+                           change_path: edit_steps_nino_path(subject: 'client'),
+                           show: true
```

Comment:
I don't think the guard on line 33 is required. If you remove the show: true  argument it would not show when the nino value was null. (Because of line 76)